browser/resources/brave_new_tab_page_refresh/components/background/background.tsx

```diff
@@ -0,0 +1,170 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+
+import { SponsoredRichMediaEventType, SponsoredImageBackground } from '../../models/backgrounds'
+import { useAppState, useAppActions } from '../context/app_model_context'
+import { openLink } from '../common/link'
+import { loadImage } from '../../lib/image_loader'
+
+import { style } from './background.style'
+
+export function Background() {
+  const currentBackground = useAppState((s) => s.currentBackground)
+
+  function renderBackground() {
+    if (!currentBackground) {
+      return <ColorBackground colorValue='transparent' />
+    }
+
+    switch (currentBackground.type) {
+      case 'brave':
+      case 'custom':
+      case 'sponsored-image':
+        return <ImageBackground url={currentBackground.imageUrl} />
+      case 'sponsored-rich-media':
+        return <SponsoredRichMediaBackground background={currentBackground} />
+      case 'solid':
+      case 'gradient':
+        return <ColorBackground colorValue={currentBackground.cssValue} />
+    }
+  }
+
+  return (
+    <div data-css-scope={style.scope}>
+      {renderBackground()}
+    </div>
+  )
+}
+
+function ColorBackground(props: { colorValue: string }) {
+  React.useEffect(() => {
+    setBackgroundVariable(props.colorValue)
+  }, [props.colorValue])
+
+  return <div className='color-background' />
+}
+
+function setBackgroundVariable(value: string) {
+  if (value) {
+    document.body.style.setProperty('--ntp-background', value)
+  } else {
+    document.body.style.removeProperty('--ntp-background')
+  }
+}
+
+function ImageBackground(props: { url: string }) {
+  // In order to avoid a "flash-of-unloaded-image", load the image in the
+  // background and only update the background CSS variable when the image has
+  // finished loading.
+  React.useEffect(() => {
+    loadImage(props.url).then((loaded) => {
+      if (loaded) {
+        setBackgroundVariable(`url(${CSS.escape(props.url)})`)
```

Comment:
`loadImage` is only checking whether the image loads, it doesn't necessarily cache it so I suspect we load it twice